File: Aishagojo/count_zeros.js
 // Write a program that takes an integer as input and return an integer with reversed digit
 // ordering.

const readline = require('readline');


const rl = readline.createInterface({
    input: process.stdin,
    output: process.stdout
});

function reverseInteger(n) {
    let reversed = 0;
    let num = Math.abs(n);

    
    while (num > 0) {
        reversed = reversed * 10 + (num % 10);
        num = Math.floor(num / 10);
    }

    
    return n < 0 ? -reversed : reversed;
}


rl.question("Enter an integer: ", (input) => {
    const num = parseInt(input, 10);
    const result = reverseInteger(num);
    console.log("Reversed integer:", result);
    rl.close(); 
});

File: IthielKyama/count_zeros.py
# Program that takes an integer as input and returns true if the input is a power of two using recursion

def function(n):
    if n <= 0:
        return False
    elif n == 1:
        return True
    else:
        return n % 2 == 0 and function(n // 2)


print(function(4))
